### PR TITLE
Read schema and add username to the basic form

### PR DESCRIPTION
**Pull request by andresmgot**
Part 1 of 2

Ref #1182 

With this PR, the dashboard obtains the schema from the chartsvc (requires https://github.com/helm/monocular/pull/645) and pass it to the `DeploymentForm` component to parse it.

As an example, I have hardcoded the basic parameter `username` to show the proposed architecture.

I have defined a basic form parameter as:

```js
{
  name: username, // Name of the basic parameter
  path: account.username, // Path of the parameter in the JSON Object (e.g. {account: {username: "foo"} }
  value: user // Current value of the parameter
}
```

Result:

Screenshot from 2019-10-01 16-11-47

### CODE REVIEW

File: dashboard/src/actions/repos.test.tsx [hunk 1]
```diff
@@ -68,7 +68,10 @@ const actionTestCases: ITestCase[] = [
 actionTestCases.forEach(tc => {
   describe(tc.name, () => {
     it("has expected structure", () => {
-      const actionResult = tc.args ? tc.action.call(null, ...tc.args) : tc.action.call(null);
+      const actionResult =
```
Comment (andresmgot) [inline dashboard/src/actions/repos.test.tsx]: _this code was failing in my tests but it's unrelated to this PR_

File: dashboard/src/components/DeploymentForm/DeploymentForm.tsx [hunk 2]
```diff
@@ -85,14 +89,19 @@ class DeploymentForm extends React.Component<IDeploymentFormProps, IDeploymentFo
...
     if (!this.state.valuesModified) {
       if (version) {
-        this.setState({ appValues: nextProps.selected.values });
+        this.setState({ appValues: nextProps.selected.values || "" });
       }
     }
+
+    if (nextProps.selected.schema) {
+      this.setState({ basicFormParameters: retrieveBasicFormParams(nextProps.selected.schema) });
+    }
```
Comment (absoludity) [inline dashboard/src/components/DeploymentForm/DeploymentForm.tsx]: What version of react are we using? This should be done in https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops , as componentWillReceiveProps is being removed.

File: dashboard/src/components/DeploymentForm/DeploymentForm.tsx [hunk 3]
```diff
@@ -210,7 +217,10 @@ class DeploymentForm extends React.Component<IDeploymentFormProps, IDeploymentFo
             <Tab>Advanced</Tab>
           </TabList>
           <TabPanel>
-            <BasicDeploymentForm />
+            <BasicDeploymentForm
+              params={Object.values(this.state.basicFormParameters)}
```
Comment (absoludity) [inline dashboard/src/components/DeploymentForm/DeploymentForm.tsx]: This is the conversion from map to array that I'm not sure we need, as above.

File: dashboard/src/reducers/charts.ts [hunk 4]
```diff
@@ -75,6 +78,8 @@ const chartsReducer = (state: IChartState = initialState, action: ChartsAction):
       return { ...state, selected: chartsSelectedReducer(state.selected, action) };
     case getType(actions.charts.selectValues):
       return { ...state, selected: chartsSelectedReducer(state.selected, action) };
+    case getType(actions.charts.selectSchema):
+      return { ...state, selected: chartsSelectedReducer(state.selected, action) };
```
Comment (absoludity) [inline dashboard/src/reducers/charts.ts]: I don't know the context here, but the above 10 lines look like they could avoid the redundancy with:
```
case getType(actions.charts.resetChartVersion):
case getType(actions.charts.selectReadme):
case getType(actions.charts.errorReadme):
case getType(actions.charts.errorChart):
case getType(actions.charts.selectValues):
case getType(actions.charts.selectSchema):
  return { ...state, selected: chartsSelectedReducer(state.selected, action) };
```

File: dashboard/src/shared/schema.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { IBasicFormParam } from "./types";
+
+export function retrieveBasicFormParams(schema: any) {
+  // TBD
```
Comment (absoludity) [inline dashboard/src/shared/schema.ts]: Thanks for leaving this out of the PR to make the diff more manageable. I'll have a quick break before reviewing the next :)

File: dashboard/src/components/DeploymentForm/DeploymentForm.tsx (hunk 2 above)
Comment (andresmgot) [reply dashboard/src/components/DeploymentForm/DeploymentForm.tsx]: We are using `^16.2.0` but yes, we will need to update that hook (we are using it in many places). I will create a separate issue for that.

Comment (andresmgot) [reply dashboard/src/components/DeploymentForm/DeploymentForm.tsx]: https://github.com/kubeapps/kubeapps/issues/1200

**Pull request merged**